README.md

```diff
@@ -1,30 +1,80 @@
-```bash
-./gradlew build
-```
+[ˈbʒɛ̃n͇ʧ̑ɨk]
 
-## Testing
-Tests can be run with
-```bash
-./test_all.sh
-```
+A simple programming language with very *polished* syntax.
+
+Implemented by students at Theoretical Computer Science Department of Jagiellonian University
+as the project for *Compilers* course in the 2022/23 winter semester.
+
+The language does not really have anything out of ordinary and it's mostly C-like, with a few somewhat interesting (compared to C) features:
+
+- nested functions able to access outer variables,
+- dynamically allocated arrays with automatic reference counting,
+- generator functions, allowing to pause the execution and pass a value to the caller.
+
+The compiler is written from scratch and produces x86_64 assembly code.
+
+## Example program
 
-## Formatting
-Code *must* be formatted with `ktlint`. CI will ensure this. You can format your code with
-```bash
-./gradlew ktlintFormat
 ```
+czynność f(x: Liczba) -> Liczba {
+    jeśli (x % 2 == 0)
+        zwróć x / 2
+    wpp
+        zwróć 3 * x + 1
+}
+
+czynność główna() {
+    zm x: Liczba = wczytaj()
+    zm k: Liczba = 0
 
-## Linting
-Linter for this repo is `ktlint`. To view issues with your code you can run:
-```bash
-./gradlew ktlint
+    dopóki (x != 1) {
+        x = f(x)
+        k = k + 1
+    }
+
+    powiedz(„Liczba kroków:”)
+    napisz(k)
+}
 ```
-Any issues with the code will be treated as errors by CI.
 
-## Commits
-Use [imperative commit message convention](https://cbea.ms/git-commit).
+## What?
+
+The following table lists all of the language's keywords and their corresponding meanings.
+
+Keyword    | Meaning             | Keyword      | Meaning               | Keyword  | Meaning
+-----------|---------------------|--------------|-----------------------|----------|--------------------
+`zm`       | variable            | `zakończ`    | finish; *return unit* | `Liczba` | number; *integer*
+`wart`     | value               | `zewnętrzna` | external              | `Czy`    | whether; *boolean*
+`stała`    | constant            | `jako`       | as                    | `Nic`    | nothing; *unit*
+`jeśli`    | if                  | `ciąg`       | sequence; *array*     | `prawda` | truth; *true*
+`zaś gdy`  | and when; *else if* | `dla`        | for                   | `fałsz`  | falsity; *false*
+`wpp`      | otherwise; *else*   | `wewnątrz`   | inside; *in*          | `nic`    | nothing; *unit*
+`dopóki`   | while               | `długość`    | length                | `nie`    | not
+`przerwij` | stop; *break*       | `przekaźnik` | relay; *generator*    | `lub`    | or
+`pomiń`    | skip; *continue*    | `przekaż`    | pass; *yield*         | `oraz`   | and
+`czynność` | function            | `otrzymując` | receiving; *for*      | `wtw`    | if and only if
+`zwróć`    | return              | `od`         | from; *in*            | `albo`   | either; *xor*
+
+(Note that the characters such as `ą` and `ź` cannot be replaced.)
```

Comment:
```suggestion
(Note that characters such as `ą` and `ź` cannot be replaced.)
```

Reply:
It is "the characters", because we refer to those in the keywords in this language, not all `ą`s and `ź`s in the universe.